bitmind/validator/cache/base_cache.py

```diff
@@ -129,10 +131,27 @@ def _check_compressed_cache_size(self) -> None:
             oldest_file.unlink()
             total_size -= file_size
 
+    def _prune_extracted_cache(self) -> None:
+        """Check extracted cache size and remove oldest files if over limit."""
+        total_size = sum(f.stat().st_size for f in self._get_cached_files())
+        bt.logging.info(f"Extracted cache size: {total_size / (1024*1024*1024):.2f} GB [{self.cache_dir}]")
+        while total_size > self.max_extracted_size_bytes:
+            extracted_files = self._get_cached_files()
+            if not extracted_files:
+                break
+
+            oldest_file = min(extracted_files, key=lambda f: f.stat().st_mtime)
+            file_size = oldest_file.stat().st_size
+
+            bt.logging.info(f"Removing {oldest_file.name} to stay under size limit")
+            oldest_file.unlink()
+            total_size -= file_size
+
```

Comment:
moved file removal out of sample, implemented the same way as for compressed data now